server/src/main.rs

```diff
@@ -1,17 +1,110 @@
+use futures_lite::stream::StreamExt;
+
 use crate::model::QueryRoot;
 use crate::merkle::get_new_merkle_tree;
 use crate::routes::{graphql_handler, graphql_playground, health};
 use async_graphql::{EmptySubscription, Schema};
 use axum::{extract::Extension, routing::get, Router, Server};
+use lapin::publisher_confirm::Confirmation;
 use std::sync::Arc;
 use futures::lock::Mutex;
 use model::MutationRoot;
+use tokio_postgres::NoTls;
+use lapin::{
+    options::*, types::FieldTable, BasicProperties, Connection, ConnectionProperties,
+};
 mod model;
 mod routes;
 mod merkle;
 
 #[tokio::main]
 async fn main() {
+    /////////////// experimental postgresql integration ///////////////
+    // Connect to the database.
+    let (client, connection) = tokio_postgres::connect("host=localhost user=dev dbname=arcpay", NoTls).await.unwrap();
+
+    // The connection object performs the actual communication with the database,
+    // so spawn it off to run on its own.
+    tokio::spawn(async move {
+        if let Err(e) = connection.await {
+            eprintln!("connection error: {}", e);
```

Comment:
Does this ever actually execute? I read that async "code will not be run immediately, but will only be evaluated when the returned future is `.await`ed." (read that [here](https://doc.rust-lang.org/std/keyword.async.html). Does `tokio::spawn` actually spawn the thread and await the future?

Reply:
yup, and since there is an await in inside the spawn, the future executes but in a different thread.